src/TelemetryChannels/ServerTelemetryChannel/Shared/SamplingTelemetryProcessor.cs

```diff
@@ -139,72 +157,145 @@ public string IncludedTypes
+            bool isSampledIn = SamplingScoreGenerator.GetSamplingScore(item) < samplingPercentage;
+
+            TrackSamplingRate(samplingPercentage);
+
+            if (isSampledIn)
+            {
+                this.Next.Process(item);
             }
+            else
+            { 
+                if (TelemetryChannelEventSource.Log.IsVerboseEnabled)
+                {
+                    TelemetryChannelEventSource.Log.ItemSampledOut(item.ToString());
+                }
 
-            this.Next.Process(item);
+                TelemetryDebugWriter.WriteTelemetry(item, this.GetType().Name);
+            }
         }
 
-        private bool IsSampledIn(ITelemetry telemetry)
+        private bool IsSamplingApplicable(Type telemetryItemType)
         {
-            // check if telemetry supports sampling
-            var samplingSupportingTelemetry = telemetry as ISupportSampling;
+            var excludedTypesHashSetRef = this.excludedTypesHashSet;
+            var includedTypesHashSetRef = this.includedTypesHashSet;
 
-            if (samplingSupportingTelemetry == null)
+            if (excludedTypesHashSetRef.Count > 0 && excludedTypesHashSetRef.Contains(telemetryItemType))
+            {
+                return false;
+            }
+
+            if (includedTypesHashSetRef.Count > 0 && ! includedTypesHashSetRef.Contains(telemetryItemType))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private void TrackSamplingRate(double samplingPercentage)
+        {
+            Metric samplingMetric = this.samplingRateMetric;
+            if (samplingMetric == null)
             {
-                return true;
+                MetricManager metricManager = this.metricManager;
+                if (metricManager == null)
+                {
+                    return;
+                }
+
+                // There is an edge case where there may be several sampling processors in the pipeline.
+                // To account for that, the aggregated metric documents will be marked with sufficinet info to differentiate the sampling rates.
+                // In general, if the user is only interested in whether the sampling rate is 100% or not 100% across, these properties my be ignored.
+                samplingMetric = metricManager.CreateMetric(SamplingRateMetricName,
+                                                            new Dictionary<string, string>()
+                                                            {
+                                                                [UniqueProcessorIdMetricPropertyName] = this.uniqueProcessorId,
+                                                                [IncludedTypesMetricPropertyName] = this.IncludedTypes ?? "null",
+                                                                [ExcludedTypesMetricPropertyName] = this.ExcludedTypes ?? "null",
```

Comment:
I suggest you do metric for each *type* of telemetry being sampled. If it isn't sampled, no data is "tracked". That way your time series won't be "jumpy" meaning the set of dimensions will not shift on you all of a sudden as you (or somebody else) updates included/excluded types.